src/Components/Header/Header.jsx

```diff
@@ -33,13 +46,25 @@ export const Header = () => {
                 </div>
               )}
             </Link>
-            <i id="toggle-theme" className="fas fa-moon icon icon-color"></i>
+            <i
+              onClick={toggleLightDarkTheme}
+              className={`${
+                theme === "light" ? "fas fa-moon" : "fas fa-sun"
+              } icon icon-color`}
+            ></i>
           </div>
         </div>
         {location.pathname !== "/" &&
```

Comment:
Path checking is a repetitive task that you have to do in this approach, If you just create a layout component you can directly pass which route or page you are loading from the props.

Reply:
Never thought about this like that i will try it out